### PR TITLE
perf: alias scanning in files

### PR DESCRIPTION
**Pull request by jazanne**
Improve performance of alias scanning in files in two ways

1. cache the results of `doublestar.FilepathGlob(glob)` so that consecutive calls with same pattern can use results
2. ~swap out golang regexp with https://github.com/wasilibs/go-re2~

With just change 1 I still found the tools running over 30 min (i stopped it before it finished) in one repo, but by adding change 2 it cut the time to under 10 min. Similarly, just implementing change 2 saw run times over 20 min (i stopped it before it finished).

This isn't a total fix for performance of scanning, depending on size of repo and wildcard patterns used, but it certainly helps.

UPDATE:
pulling back on the re2 usage since the build fails - we can revisit that later

### CODE REVIEW

File: aliases/testdata/wild/nested-wild/another/another/alias_test.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+EVEN_WILDER = 'wildFlag'
```
Comment (ld-kyee) [inline aliases/testdata/wild/nested-wild/another/another/alias_test.txt:1]: 😂

**Review by ld-kyee — APPROVED** [review]
Can't say I understand entirely how this all works but the code seems 💯

**Review by mmrj — APPROVED** [review]
Approving to allow work to proceed. The Docs team is codeowners of `*.md` files in this repo, so are autotagged here, but I did not review the vendor `*.md` files carefully.

**Comment by jazanne** [comment]
unfortunately the build is failing now that we're using re2, so i'm going to revert that change

**Pull request closed without merging**